akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/ReplayingEvents.scala

```diff
@@ -116,10 +116,8 @@ private[akka] final class ReplayingEvents[C, E, S](
 
             def handleEvent(event: E): Unit = {
               eventForErrorReporting = OptionVal.Some(event)
-              state = state.copy(
-                seqNr = repr.sequenceNr,
-                state = setup.eventHandler(state.state, event),
-                eventSeenInInterval = true)
+              state = state.copy(seqNr = repr.sequenceNr)
+              state = state.copy(state = setup.eventHandler(state.state, event), eventSeenInInterval = true)
```

Comment:
local val for the intermediate state would "feel" better (I don't think anything can go wrong like this though).

Reply:
A local val won't work because you must update state var, so the seqNr can be read inside eventHandler

Reply:
yes, the `currentSequenceNumber` is grabbing it from the `var state`, and that may be used from inside setup.eventHandler

Reply:
Ah, I see, I thought it was just about the state passed into the vent handler but ofc.